plugins/module_utils/core.py

```diff
@@ -375,6 +376,8 @@ def scrub_none_parameters(parameters):
     for k, v in parameters.items():
         if isinstance(v, dict):
             clean_parameters[k] = scrub_none_parameters(v)
+        elif descend_into_lists and isinstance(v, list):
+            clean_parameters[k] = [scrub_none_parameters(vv) if isinstance(vv, dict) else vv for vv in v]
```

Comment:
Although I don't think boto/boto3 generally use them this won't handle lists of lists...

Not a blocker, just thinking out loud.